### ISSUE TITLE
C# parser reads "$" as "Interpolated string expression" which is not correct

### ISSUE THREAD

**Issue by Khaos66** [doxygen/doxygen#9845]
**Describe the bug**
With this C# code the method is not recognized correctly
```C#
namespace Sample.NS
{
    public class MyClass
    {
        Regex pattern = new(@".$");

        public void Fu() { }
    }
}
```

**Expected behavior**
In `readme.md` I need to use `@ref Sample.NS.MyClass.Fu` as a reference

**To Reproduce**
[DoxygenSample.zip](https://github.com/doxygen/doxygen/files/10701220/DoxygenSample.zip)
In this sample the method before the `".$"` string is recognized, but the second method after it is not.

**Version**
doxygen 1.9.6 has this issue

**Comment by Khaos66** [comment]
@albert-github I hope the issue is setup correctly

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #9847

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.7.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
